### DISCUSSION TITLE
How to create a file server using nicegui ?

### DISCUSSION THREAD

**Discussion by levywang** [zauberzeug/nicegui#2766] [Q&A] [answered]
### Question

Hello, esteemed author. I need to use nicegui to create a file server that can download directories and files from the server. However, there isn't an example to teach me how to create it. Could you offer me some help if you have time? Thank you very much.

**Comment by rodja** [comment, marked as answer]
Have a look at https://github.com/zauberzeug/nicegui/tree/main/examples/local_file_picker

**Reply by levywang** [reply]
Thank you very much for your response. Prior to this, I have already reviewed the example you mentioned, which is a demo for file uploading. I know how to upload files, but I'm not sure how to bind directories and display them from nicegui. Could you provide me with some hints? Thank you.

**Reply by rodja** [reply]
The example shows how to chose files from a "local" directory. "local" means on the server, not from the browser. So it's not about uploading but as you requested, browsing and picking files on the system where NiceGUI is executed.

**Reply by levywang** [reply]
I understand now, thank you very much for your help.
